### PR TITLE
[API] Add affected files of commits to commit struct

### PR DESCRIPTION
**Pull request by 6543**
finish #12413 ...

TODO:
 * [x] Extend Tests

### CODE REVIEW

File: integrations/api_repo_git_commits_test.go [hunk 1]
```diff
@@ -93,6 +104,7 @@ func TestAPIReposGitCommitListDifferentBranch(t *testing.T) {
 	var apiData []api.Commit
 	DecodeJSON(t, resp, &apiData)
 
-	assert.Equal(t, 1, len(apiData))
+	assert.Len(t, apiData, 1)
 	assert.Equal(t, "f27c2b2b03dcab38beaf89b0ab4ff61f6de63441", apiData[0].CommitMeta.SHA)
+	compareCommitFiles(t, []string{"readme.md"}, apiData[2].Files)
```
Comment (6543) [inline integrations/api_repo_git_commits_test.go]:
```suggestion
	compareCommitFiles(t, []string{"readme.md"}, apiData[0].Files)
```

**Comment by lunny** [comment]
Fix #12412

**Review by lunny — APPROVED** [review]

File: modules/convert/git_commit.go [hunk 2]
```diff
@@ -131,6 +131,20 @@ func ToCommit(repo *models.Repository, commit *git.Commit, userCache map[string]
...
+		return nil, err
+	}
+	affectedFiles := append(fileStatus.Added, fileStatus.Removed...)
+	affectedFiles = append(affectedFiles, fileStatus.Modified...)
+	affectedFileList := make([]*api.CommitAffectedFiles, len(affectedFiles))
+	for i := range affectedFiles {
+		affectedFileList[i] = &api.CommitAffectedFiles{
+			Filename: affectedFiles[i],
+		}
+	}
```
Comment (zeripath) [inline modules/convert/git_commit.go]: We can iterate through these a single time:

```suggestion
	affectedFileList := make([]*api.CommitAffectedFiles, 0, len(fileStatus.Added)+len(fileStatus.Removed)+len(fileStatus.Modified))
	for _, files := range [][]string{fileStatus.Added, fileStatus.Removed, fileStatus.Modified} {
		for _, filename := range files {
			affectedFileList = append(affectedFileList, &api.CommitAffectedFiles{
				Filename: filename,
			})
		}
	}
```

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Pull request merged**